src/support/z_media_preview.erl

```diff
@@ -83,8 +83,15 @@ convert_1(false, _InFile, _OutFile, _Mime, _FileProps, _Filters) ->
 convert_1(ConvertCmd, InFile, OutFile, Mime, FileProps, Filters) ->
     OutMime = z_media_identify:guess_mime(OutFile),
     {EndWidth, EndHeight, CmdArgs} = cmd_args(FileProps, Filters, OutMime),
-    z_utils:assert(EndWidth  < ?MAX_WIDTH, image_too_wide),
-    z_utils:assert(EndHeight < ?MAX_HEIGHT, image_too_high),
+    convert_2(EndWidth, EndHeight, CmdArgs, ConvertCmd, InFile, OutFile, Mime, FileProps).
+
+convert_2(EndWidth, _EndHeight, _CmdArgs, _ConvertCmd, _InFile, _OutFile, _Mime, _FileProps) 
+    when EndWidth > ?MAX_WIDTH ->
+    {error, image_too_wide};
+convert_2(_EndWidth, EndHeight, _CmdArgs, _ConvertCmd, _InFile, _OutFile, _Mime, _FileProps) 
+    when EndHeight > ?MAX_HEIGHT ->
+    {error, image_too_wide};
```

Comment:
Or shall we both make `image_size` ?
Less cases to check on, and if there is some rotation then the width/height is unclear anyway.